src/create_agent/check_agent.jl

```diff
@@ -0,0 +1,35 @@
+"""
+
+Function for checking the structure of the agent
+"""
+function check_agent(agent::Agent)
+
+   if length(agent.shared_parameters)> 0
+
+    ## Check for the same derived parameter in multiple shared parameters 
+    #Get out the derived parameters of all shared parameters
+    vector_of_derived_parameters = [agent.shared_parameters[i].derived_parameters for i in unique(keys(agent.shared_parameters))]
+    #combine them to one list
+    all_derived_parameters = [y for v in vector_of_derived_parameters for y in v]
+    #check for duplicate names
+    if length(all_derived_parameters) > length(unique(all_derived_parameters ))
+        #Throw an error
+        throw(
+            ArgumentError(
+                "The same derived parameter has two shared parameters",
```

Comment:
Sørg for at skriv hvilken derived parameter der kommer to gange.
Man kan sætte en variabel in i en string ved at skrive "blablaba $variable blabla".
Man kan måske bruge countmap til at tælle hvor mange af hver der er, i stedet for at bruge unique; så kan erroren sige alle de derived parameters der kommer to tange. 
https://discourse.julialang.org/t/how-to-find-duplicate-rows-in-string-array-vec/49182
Jeg synes vi skal undgå at kalde noget en 'derived parameter' i error messages osv. Bare sig at nogel shared parmeters specifyer de samme parameters eller sådn noget

Reply:
countmap er fra StatsBase, tænker ikke vi skal have den som dependency? og ja tjak

Reply:
jeg har lavet en lidt anden løsning på det uden pakker :))